Improved shard allocation awareness attributes documentation (#8268)
opensearch-project/documentation-website: _tuning-your-cluster/index.md

## (Advanced) Step 6: Configure shard allocation awareness or forced awareness

To further fine-tune your shard allocation, you can set custom node attributes for shard allocation awareness or forced awareness.

### Shard allocation awareness

You can set custom node attributes on OpenSearch nodes to be used for shard allocation awareness. For example, you can set the `zone` attribute on each node to represent the zone in which the node is located. You can also use the `zone` attribute to ensure that primary shard and its replica shards are allocated in a balanced manner across available distinct zones, for example, maximum shard copies per zone would equal `ceil (number_of_shard_copies/number_of_distinct_zones)`.

> Shard allocation awareness attempts to separate primary and replica shards across multiple zones. However, if only one zone is available (such as after a zone failure), OpenSearch allocates replica shards to the only remaining zone. If your index has total 5 shard copies (1 primary and 4 replicas) and nodes in only 3 distinct zones, then OpenSearch will not allocate more than 2 shards per zone, so you need at least 2 nodes in 2 zones each and at least 1 node in the 3rd zone to allocate all 5 shard copies. 

Reviewer: @patelsmit32123: I think breaking up this information will make it more digestible. What do you think of this small rewrite?

```suggestion
Shard allocation awareness attempts to separate primary and replica shards across multiple zones. However, if only one zone is available (such as after a zone failure), OpenSearch allocates replica shards to the only remaining zone. For example, if your index has a total 5 shard copies (1 primary and 4 replicas) and nodes in 3 distinct zones, then OpenSearch will perform the following to allocate all 5 shard copies: 

- Allocate less than 2 shards per zone, which means you'll need at least 2 nodes in 2 zones.
- Allocate the last node in the 3rd zone. 
```

Author: makes sense to me, thanks

Author: - Allocate the last node in the 3rd zone. => - Allocate the last shard in the 3rd zone, at least 1 node needed in the 3rd zone.

> If you have 3 nodes in the 1st zone and 1 node each in every remaining zone, then OpenSearch will only allocate 2,1,1 shard copies leaving the 5th copy unallocated.

Author: yes, two shard copies can't be placed on the same node, there is a allocation decider for the same too

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/routing/allocation/decider/SameShardAllocationDecider.java

## (Advanced) Step 6: Configure shard allocation awareness or forced awareness

To further fine-tune your shard allocation, you can set custom node attributes for shard allocation awareness or forced awareness.

### Shard allocation awareness

> You can set custom node attributes on OpenSearch nodes to be used for shard allocation awareness. For example, you can set the `zone` attribute on each node to represent the zone in which the node is located. You can also use the `zone` attribute to ensure that primary shard and its replica shards are allocated in a balanced manner across available distinct zones, for example, maximum shard copies per zone would equal `ceil (number_of_shard_copies/number_of_distinct_zones)`.

Reviewer 2: ```suggestion
You can set custom node attributes on OpenSearch nodes to be used for shard allocation awareness. For example, you can set the `zone` attribute on each node to represent the zone in which the node is located. You can also use the `zone` attribute to ensure that the primary shard and its replica shards are allocated in a balanced manner across available, distinct zones. For example, maximum shard copies per zone would equal `ceil (number_of_shard_copies/number_of_distinct_zones)`.
```

> Shard allocation awareness attempts to separate primary and replica shards across multiple zones, because two shard copies can not be placed on the same node. When only one zone is available, such as after a zone failure, OpenSearch allocates replica shards to the only remaining zone. For example, if your index has a total 5 shard copies (1 primary and 4 replicas) and nodes in 3 distinct zones, then OpenSearch will perform the following to allocate all 5 shard copies: 

Reviewer 2: ```suggestion
Shard allocation awareness attempts to separate primary and replica shards across multiple zones because 2 shard copies cannot be placed on the same node. When only 1 zone is available, such as after a zone failure, OpenSearch allocates replica shards to the only remaining zone. For example, if your index has a total of 5 shard copies (1 primary and 4 replicas) and nodes in 3 distinct zones, then OpenSearch will perform the following to allocate all 5 shard copies: 
```

> - Allocate less than 2 shards per zone, which means you'll need at least 2 nodes in 2 zones.

Reviewer 2: ```suggestion
- Allocate fewer than 2 shards per zone, which will require at least 2 nodes in 2 zones.
```

> - Allocate the last shard in the 3rd zone, with at least one node needed in the 3rd zone.

Reviewer 2: ```suggestion
- Allocate the last shard in the third zone, with at least 1 node needed in the third zone.
```

> Alternatively, if you have 3 nodes in the 1st zone and 1 node each remaining zone , then OpenSearch will allocate:

Reviewer 2: ```suggestion
Alternatively, if you have 3 nodes in the first zone and 1 node in each remaining zone, then OpenSearch will allocate:
```

> - 2 shard copies in the 1st zone

Reviewer 2: ```suggestion
- 2 shard copies in the first zone.
```

> - 1 shard copy in the remaining two zones

Reviewer 2: ```suggestion
- 1 shard copy in the remaining 2 zones.
```

The final shard copy will remain unallocated due to the lack of nodes. 

> With shard allocation awareness, if the nodes in one of your zones fail, you can be assured that your replica shards are spread across your other zones, adding a layer of fault tolerance to ensure your data survives zone failures.

Reviewer 2: ```suggestion
With shard allocation awareness, if the nodes in one of your zones fail, you can be assured that your replica shards are spread across your other zones, adding a layer of fault tolerance to ensure that your data survives zone failures.
```

}
```

> You can also use multiple attributes for shard allocation awareness by providing the attributes as a comma separated string, for example `zone,rack`.

Reviewer 2: ```suggestion
You can also use multiple attributes for shard allocation awareness by providing the attributes as a comma-separated string, for example, `zone,rack`.
```
